### DISCUSSION TITLE
Difference between edger rscripts in directory "code" to directory "src"

### DISCUSSION THREAD

**Discussion by gsk7757** [biostars/biostar-handbook-issues#308] [Q&A]
Hello, 
I tried running rscript (edger.r) from the directory "code", which was mentioned in the book RNA-seq by example. This gave me a differential expression list with 29 upregulated genes and 218 downregulated genes. However, when I ran rscript (edger.r) from the directory "src" from the biostar work flows, it gave me 90 upregulated genes and 28 downregulated genes. When I used deseq2 scripts from both the directories, the result is 92 upregulated genes and 26 downregulated genes. May I know why there is a significant difference in the results between the edger scripts and what might have caused the difference?

**Comment by ialbert** [comment]
the new code uses a different method `glm` vs the so-called `classic` method within edge.r

Also I think it preemptively filters out more of the low-expression genes, hence reducing the cost of multiple comparisons.

With the new code, you can select the method with the `-m` flag as either `glm` or `classic`

**Reply by ialbert** [reply]
I will also say that I like the classic method, the results are fewer but more robust and sometimes a cleaner message

it used to be that deseq2 produced more hits and edger produced fewer but more selective results.

**Reply by gsk7757** [reply]
Thank you so much for the clarification. I will stick with the rscript that uses the classic method. May I know when I use this data for the process of publication, how do I cite the book RNA-seq by example? As of now, I have mentioned in the methods sections as "The computer setup for the analysis of the RNA-seq and scripts for the edgeR were from RNA-seq by Example, a volume from the Biostar Handbook Collection". Please correct me if I am wrong.

**Reply by ialbert** [reply]
yes that sounds good,  "RNA-Seq by Example" volume of the Biostar Handbook ISBN: 978-0-578-80435-4

Only the BIostar Handbook has an ISBN number at this time.

**Reply by gsk7757** [reply]
Thank you so much
